server/channels/app/web_broadcast_hooks.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	pUtils "github.com/mattermost/mattermost/server/public/utils"
+	"github.com/mattermost/mattermost/server/v8/channels/app/platform"
+)
+
+const (
+	broadcastAddMentions  = "add_mentions"
+	broadcastAddFollowers = "addfollowers"
+)
+
+func (s *Server) makeBroadcastHooks() map[string]platform.BroadcastHook {
+	return map[string]platform.BroadcastHook{
+		broadcastAddMentions:  &addMentionsBroadcastHook{},
+		broadcastAddFollowers: &addFollowersBroadcastHook{},
+	}
+}
+
+type addMentionsBroadcastHook struct{}
+
+func (h *addMentionsBroadcastHook) ShouldProcess(msg *model.WebSocketEvent, webConn *platform.WebConn, args map[string]any) bool {
+	if msg.EventType() != model.WebsocketEventPosted {
+		return false
+	}
+
+	mentions, ok := args["mentions"].(model.StringArray)
+	if !ok {
+		mlog.Warn("Invalid mentions value passed to addMentionsBroadcastHook", mlog.Any("mentions", args["mentions"]))
+		return false
+	}
+
+	// This hook will only modify the event if the current user is mentioned by the post
+	return pUtils.Contains[string](mentions, webConn.UserId)
+}
+
+func (h *addMentionsBroadcastHook) Process(msg *model.WebSocketEvent, webConn *platform.WebConn, args map[string]any) *model.WebSocketEvent {
+	mentions, ok := args["mentions"].(model.StringArray)
+	if !ok {
+		mlog.Warn("Invalid mentions value passed to addMentionsBroadcastHook", mlog.Any("mentions", args["mentions"]))
```

Comment:
Is there a reason for not returning a proper error instead?

Reply:
Not particularly. I think I had a good reason before, but I can't think of it now. I'll change that